org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend

```diff
@@ -296,4 +314,13 @@ class CollectionTestCase extends AbstractWollokInterpreterTestCase {
 		}'''.interpretPropagatingErrors
 	}
 	
+	@Test
+	def void sumNoArgs() {
+		'''
+		program p {
+			«instantiateCollectionAsNumbersVariable»
+			assert.equals(34, numbers.sum())
```

Comment:
We certainly can, but now that you mention it... min(closure) and
max(closure), which are the base for min() and max() respectively, use
absolute(closure, criteria) as their base. This method doesn't fail on
empty lists, but returns null. I'd rather have min and max (either version
of each one) fail, what do you think?

The case of sum() or sum(closure) is much simpler, of course.

On Wed, Jun 8, 2016 at 7:30 PM, Franco Leonardo Bulgarelli <
notifications@github.com> wrote:

> In
> org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/CollectionTestCase.xtend
> <https://github.com/uqbar-project/wollok/pull/718#discussion_r66352475>:
>
> > @@ -296,4 +314,13 @@ class CollectionTestCase extends AbstractWollokInterpreterTestCase {
> >  		}'''.interpretPropagatingErrors
> >  	}
> >  	
> > +	@Test
> > +	def void sumNoArgs() {
> > +		'''
> > +		program p {
> > +			«instantiateCollectionAsNumbersVariable»
> > +			assert.equals(34, numbers.sum())
>
> @matifreyre <https://github.com/matifreyre> Could we add tests for the
> edge case of empty and unitary collections?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/wollok/pull/718/files/16fef21ed649c1d7f3ec9227c7161c6b00ce587e#r66352475>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ-EZnVeSg4vytnaGOccgTm08GpdbQodks5qJ0KggaJpZM4Ixc9x>
> .
>